### PR TITLE
Feature/lu 6203 convert sdc to spp format

### PR DESCRIPTION
**Pull request by ranjit-das**

### CODE REVIEW

File: lambda-ingest-json-to-json/json_to_json.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import os
+import json
+import boto3
+from datetime import datetime
+from uuid import uuid1
+
+def read_file_from_s3(json_title):
+    landing_zone_bucket = os.getenv("LANDING_ZONE_BUCKET")
+    landing_zone_filename = json_title
```
Comment (dllmorgan) [inline lambda-ingest-json-to-json/json_to_json.py:9]: landing_zone_filename is being assigned but only used to pass into the S3 object call.

File: lambda-ingest-json-to-json/json_to_json.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    return body
+
+#Converts input json to output json
+def process_contributors(data):
+    try:
+        json_data = json.loads(data)
+    except Exception as error:
+        print("Invalid Json")
+        print(error)
+        raise Exception
```
Comment (dllmorgan) [inline lambda-ingest-json-to-json/json_to_json.py:23]: This raise is generating a new empty exception (i.e. no error information is passed back up to where it is caught)

File: lambda-ingest-json-to-json/handler.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+def lambda_handler(event, context):
+    lambda_name = "takeon-ingest-json-to-json"
+    print("Event:" + str(event))
+    json_title = event['detail']['requestParameters']['key']
+
+    try:
+        main(json_title)
+        print("Success: " + lambda_name)
+    except Exception as error:
```
Comment (dllmorgan) [inline lambda-ingest-json-to-json/handler.py:14]: Any exceptions are already 'double-caught' in the levels below. Not sure this exception code will be executed.

File: lambda-ingest-json-to-json/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# black==19.3b0
+# flake8==3.7.7
+# flake8-polyfill==1.0.2
+# virtualenv==16.6.1
+# PyYAML==5.1
+# requests==2.22.0
+# coverage==4.5.3
+# decorator==4.4.0
+# boto3==1.9.155
+# botocore==1.12.181
```
Comment (dllmorgan) [inline lambda-ingest-json-to-json/requirements.txt:20]: Commented out requirements can probably be removed.

File: lambda-ingest-json-to-json/tests/test_json_to_json.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
+import json_to_json
```
Comment (dllmorgan) [inline lambda-ingest-json-to-json/tests/test_json_to_json.py:1]: When I run this test locally I get an assertion error.
Also, what about tests to:

1) confirm behaviour when the source json is malformed?
2) confirm behaviour when the json is valid but expected attributes are missing?
3) confirm behaviour when json is valid and has expected attributes but no question responses?

File: lambda-ingest-json-to-json/tests/test_json_to_json.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+def test_process_contributors_validAndCompleteRSI_ValidJSONOutput():
+    survey = "0066"
```
Comment (dllmorgan) [inline lambda-ingest-json-to-json/tests/test_json_to_json.py:5]: survey 0066?

File: aws-infrastructure/ingest.tf [hunk 7]
```diff
@@ -332,6 +332,39 @@ resource "aws_lambda_function" "pcktojson-bricks" {
   )
 }
 
+# Ingest pck to json lambda
+resource "aws_lambda_function" "jsontojson" {
+
+  function_name = "${local.name_prefix}-json-to-json"
```
[the remark this answers is not in the document]
Comment (ranjit-das) [reply aws-infrastructure/ingest.tf:338]: Changed to "SDCjson-to-SPPjson"

File: lambda-ingest-json-to-json/handler.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import boto3
+import json
+from json_to_json import main
+
+
+def lambda_handler(event, context):
+    lambda_name = "takeon-ingest-json-to-json"
+    print("Event:" + str(event))
+    json_title = event['detail']['requestParameters']['key']
```
[the remark this answers is not in the document]
Comment (ranjit-das) [reply lambda-ingest-json-to-json/handler.py]: It is the file name extracted from the event. So now changed to rsi_input_file_name.

File: lambda-ingest-json-to-json/handler.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import boto3
+import json
+from json_to_json import main
+
+
+def lambda_handler(event, context):
+    lambda_name = "takeon-ingest-json-to-json"
```
[the remark this answers is not in the document]
Comment (ranjit-das) [reply lambda-ingest-json-to-json/handler.py]: Removed lambda_name.

File: lambda-ingest-json-to-json/json_to_json.py (hunk 1 above)
Comment (ranjit-das) [reply lambda-ingest-json-to-json/json_to_json.py]: un necessary assignment removed and used rsi_input_file_name insated of json_title

File: lambda-ingest-json-to-json/json_to_json.py (hunk 2 above)
Comment (ranjit-das) [reply lambda-ingest-json-to-json/json_to_json.py]: raise Exception is now passed with msg - 'Invalid Json' + error

File: lambda-ingest-json-to-json/handler.py (hunk 3 above)
Comment (ranjit-das) [reply lambda-ingest-json-to-json/handler.py]: Exception handling updated and now it logs only in this place

File: lambda-ingest-json-to-json/requirements.txt (hunk 4 above)
Comment (ranjit-das) [reply lambda-ingest-json-to-json/requirements.txt]: yes, removed.

File: lambda-ingest-json-to-json/tests/test_json_to_json.py (hunk 6 above)
Comment (ranjit-das) [reply lambda-ingest-json-to-json/tests/test_json_to_json.py]: removed...now survey comes in the input json

File: lambda-ingest-json-to-json/tests/test_json_to_json.py (hunk 5 above)
Comment (ranjit-das) [reply lambda-ingest-json-to-json/tests/test_json_to_json.py]: tests are included.

File: aws-infrastructure/ingest.tf [hunk 10]
```diff
@@ -541,7 +573,8 @@ resource "aws_cloudwatch_event_rule" "pcktojson" {
 				"eventName": ["PutObject"],
 				"requestParameters": {
 					"bucketName": ["${aws_s3_bucket.landingzone.bucket}"],
-					"key": [{"anything-but": {"prefix": "storeMBR"}}]
+					"key": [{"anything-but": {"prefix": "MBR"}}],
+                    "key": [{"anything-but": {"prefix": "RSI"}}]
```
[the remark this answers is not in the document]
Comment (ranjit-das) [reply aws-infrastructure/ingest.tf:577]: Yes with file name like storeQSL177648 now triggers pcktojson

**Pull request merged**